In [2]:
import pandas as pd
import scipy.stats as stats

df = pd.read_csv("/content/teachers_rating_data.csv")

print(df.head())

     prof  Gender  Age Tenure  Evaluation  Students  Beauty CourseLevel
0  Prof52    Male   59     No        4.38        69     3.1       Lower
1  Prof93    Male   59     No        3.12        71     4.2       Lower
2  Prof15    Male   51     No        4.83        38     4.0       Lower
3  Prof72    Male   36    Yes        3.88        79     3.4       Lower
4  Prof61  Female   47     No        3.48        56     2.8       Upper


# Q1. T-Test: Using the teachers' rating data set, does gender affect teaching evaluation rates?

In [5]:
correlation, p_value = stats.pearsonr(df['Beauty'], df['Evaluation'])

print(f"Correlation between Beauty and Evaluation: {correlation:.4f}")
print(f"P-value: {p_value:.4f}")

Correlation between Beauty and Evaluation: -0.1830
P-value: 0.0454


# Correlation Coefficient: -0.1830

This value indicates a weak negative linear relationship between 'Beauty' and 'Evaluation'.

In practical terms, this suggests that as a professor's perceived beauty score increases, their evaluation score tends to slightly decrease.

# P-value: 0.0454

Since this p-value is less than the standard significance level of 0.05, the correlation is considered statistically significant.

This means we can be reasonably confident that the weak negative relationship observed in the sample is not due to random chance and likely exists in the broader population from which the sample was drawn.

# Q2. ANOVA: Using the teachers' rating data set, does beauty score for instructors differ by age?

## Objective:
To check whether beauty scores differ significantly across different age groups.

## Hypotheses:

H₀: Mean beauty scores are equal across all age groups.

H₁: At least one group has a different mean beauty score.

## Test Used:
One-way ANOVA

In [6]:
male_eval = df[df['Gender'] == 'Male']['Evaluation']
female_eval = df[df['Gender'] == 'Female']['Evaluation']

t_stat, p_value_gender = stats.ttest_ind(male_eval, female_eval)

print(f"T-statistic: {t_stat:.4f}")
print(f"P-value: {p_value_gender:.4f}")

T-statistic: 2.2958
P-value: 0.0235


# T-statistic: 2.2958

This positive value indicates that the average 'Evaluation' score for the first group in the comparison (male professors) is higher than the average score for the second group (female professors).

#P-value: 0.0235

This p-value is less than the standard significance level of 0.05.Therefore, the result is statistically significant. We can conclude that there is a significant difference in the mean evaluation scores between male and female professors in this dataset.

# Q3. Chi-square: Using the teachers' rating data set, is there an association between tenure and gender?

# Objective:
To test whether tenure and gender are independent or associated.

# Hypotheses:

H₀: Tenure and gender are independent.

H₁: There is an association between tenure and gender.

Test Used:
Chi-square test of independence

In [7]:
contingency_table = pd.crosstab(df['Tenure'], df['Gender'])
chi2, p, dof, expected = stats.chi2_contingency(contingency_table)
print("Chi-square:", chi2)
print("P-Value:", p)

Chi-square: 0.28968284722760423
P-Value: 0.5904238409135254


# Interpretation:

fail to reject H₀ → They are independent.

# Q4. Correlation: Using the teachers' rating dataset, is teaching evaluation score correlated with beauty score?

# Objective:
To test whether there is a linear relationship between evaluation and beauty.

# Hypotheses:

H₀: There is no correlation between evaluation and beauty.

H₁: There is a correlation between evaluation and beauty.

Test Used:
Pearson Correlation Coefficient

In [9]:
tenured_eval = df[df['Tenure'] == 'Yes']['Evaluation']
non_tenured_eval = df[df['Tenure'] == 'No']['Evaluation']

t_stat_tenure, p_value_tenure = stats.ttest_ind(tenured_eval, non_tenured_eval)

print(f"T-statistic: {t_stat_tenure:.4f}")
print(f"P-value: {p_value_tenure:.4f}")

T-statistic: -0.8877
P-value: 0.3765


# T-statistic: -0.8877

This negative value indicates that the average 'Evaluation' score for tenured professors is slightly lower than the average score for non-tenured professors.

# P-value: 0.3765

This p-value is significantly greater than the standard significance level of $0.05$.Therefore, the result is not statistically significant.